preprocessor/processor.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+import os
+import glob
+import cv2 as cv
+
+allowed_file_extensions = ['.png', '.jpg', '.jpeg']
+
+# recursively load and normalize images using histogram equalization from OpenCV
+def normalize_images(inputdirectory, outputdirectory):
+    inputpath = Path(inputdirectory)
+    outputpath = Path(outputdirectory)
+    inputfiles = []
+
+    for fe in allowed_file_extensions:
+        inputfiles.extend(list(inputpath.glob('**/*' + fe)))
+
+    for f in inputfiles:
+        abs_file_path = Path.absolute(f)
+        file_path = str(abs_file_path)
+        output_file_name = get_output_file_name(f, inputpath, outputpath)
+        print('normalizing image: ' + str(f) + ' to: ' + str(output_file_name))
+
+        img_output = normalize_image(file_path)
+
+        ensure_dir_exists_for_file(output_file_name)
+
+        save_image(str(output_file_name), img_output)
+
+def normalize_image(file_path):
+        img = cv.imread(file_path)
+        img_yuv = cv.cvtColor(img, cv.COLOR_BGR2YUV)
+
+        img_yuv[:,:,0] = cv.equalizeHist(img_yuv[:,:,0])
```

Comment:
I wonder if there are some other cleanups that we should also implement here. I haven't done much research in this area yet, but [skimage.expose](http://scikit-image.org/docs/dev/api/skimage.exposure.html) has a bunch of nice utilities. If you have time, it would be great to investigate what other sorts of cleanups we could/should perform.